File: solutions/c/darts/1/darts.c
#include "darts.h"

float calculate_distance_from_center(coordinate_t coor){
  return sqrt(coor.x * coor.x + coor.y * coor.y);
}

uint8_t score(coordinate_t coor){
  float distance = calculate_distance_from_center(coor);
  if(distance > 10) return 0;
  else if(distance > 5) return 1;
  else if(distance > 1) return 5;
  return 10;
}

File: solutions/csharp/darts/1/Darts.cs
public static class Darts
{
    public static int Score(double x, double y)
        => DistanceOrigin(x, y) switch{
            <= 1 => 10,
            <= 5 => 5,
            <= 10 => 1,
            _ => 0
        };

    private static double DistanceOrigin(double x, double y)
        => Math.Sqrt(x*x+y*y);
}
